### ISSUE TITLE
User gets white screen after update is deployed

### ISSUE THREAD

**Issue by andylhansen** [react-boilerplate/react-boilerplate#1325]
Whenever I push updates to my app, users that have already used it get a white screen, and they have to clear their cache in order for the update to work? Is there some way to fix this?

I've tried following the instructions at this page: https://github.com/NekR/offline-plugin/blob/master/docs/updates.md
but I'm still having users get a white screen. It seems to mostly affect safari, so it could have something to do with safari not supporting service workers.

**Comment by mxstbr** [comment]
@NekR?

**Comment by NekR** [comment]
> It seems to mostly affect safari, so it could have something to do with safari not supporting service workers.

Safari doesn't support ServiceWorker and AppCache is disabled in `react-boilerplate` by default, so nothing could affect Safari from `offline-plugin` side here. @andylhansen what makes you feel it's `offline-plugin` related? What config of it you are using? What version?

**Comment by zpl** [comment]
Yesterday I had exactly same problem - just after upgrading react-boilerplate there is "white page" in production build. (dev working without any problems)

after some time of debugging I found a problem: **react-loader-advanced**

If I commented it out - everything works again...


you can reproduce this error:

git clone git@github.com:mxstbr/react-boilerplate
npm run setup
npm install react-loader-advanced --save

and add import Loader from 'react-loader-advanced' to HomePage for example

now it will show blank screen on production build.

**Comment by andylhansen** [comment]
I'm using version 3.1.0 of the react boilerplate.
It could be the lack of offline-plugin support is causing the issue.
I am using the boilerplate's built in production server. Perhaps the server is telling the browser to cache the files for too long, or is returning an improper 304 code?
Before I added the service worker auto-reload code that I mentioned, both safari and chrome would get the blank screen whenever I pushed updates, and they wouldn't load until I hard refreshed. With the auto reload code, chrome still shows the blank screen for a few seconds before reloading. Safari just stays on the blank screen until a hard reload.
